server/app/jobs/grid_service_scheduler_worker.rb

```diff
@@ -41,6 +41,10 @@ def check_deploy_queue
         return nil
       end
     end
+  rescue => exc
+    warn "aborting deploy on un-handled error: #{exc.message}"
+    warn exc.backtrace.join("\n") if exc.backtrace
+    service_deploy.abort! "service deploy aborted: #{exc.message}" if service_deploy
```

Comment:
Reminds me that we should figure out how to display these abort reasons to the user... doesn't happen ATM.